### PR TITLE
Add posthog Telemetry (Anonymity Logic + call duration telemetry)

### PR DESCRIPTION
**Pull request by toger5**
Fixes: https://github.com/vector-im/element-call/issues/353
Fixes: https://github.com/vector-im/element-call/issues/659
Depends on: #682

### CODE REVIEW

File: src/PosthogAnalytics.ts [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+  SignupTracker,
+  MuteCameraTracker,
+  MuteMicrophoneTracker,
+} from "./PosthogEvents";
+
+/* Posthog analytics tracking.
+ *
+ * Anonymity behaviour is as follows:
+ *
+ * - If Posthog isn't configured in `config.json`, events are not sent.
```
Comment (SimonBrandner) [inline src/PosthogAnalytics.ts:34]: Don't think we have a `config.json`?

File: src/PosthogAnalytics.ts [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+ * - If [Do Not Track](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/doNotTrack) is
+ *   enabled, events are not sent (this detection is built into posthog and turned on via the
+ *   `respect_dnt` flag being passed to `posthog.init`).
+ * - If the `feature_pseudonymous_analytics_opt_in` labs flag is `true`, track pseudonomously by maintaining
```
Comment (SimonBrandner) [inline src/PosthogAnalytics.ts:38]: Do we really use this name `feature_pseudonymous_analytics_opt_in`?

**Review by SimonBrandner — COMMENTED** [review]
Otherwise LGTM

File: src/PosthogAnalytics.ts [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+}
+
+export enum RegistrationType {
+  Guest,
+  Registered,
+}
+
+interface PlatformProperties {
+  appVersion: string;
+  appPlatform: "embedded" | "jssdk";
```
[the remark this answers is not in the document]
Comment (toger5) [reply src/PosthogAnalytics.ts:74]: for now the platform only represents which backend is used.
I think it should not be called platform then...

[the remark this answers is not in the document]
Comment (toger5) [reply src/PosthogAnalytics.ts:74]: But now that you are saying it, it makes sense that in the future the PlatformProperties should also include Android, iOS, Web, Desktop...

File: src/PosthogAnalytics.ts [hunk 4]
```diff
@@ -0,0 +1,336 @@
+/*
+Copyright 2021 The Matrix.org Foundation C.I.C.
```
Comment (dbkr) [inline src/PosthogAnalytics.ts:2]: year

File: src/PosthogAnalytics.ts [hunk 5]
```diff
@@ -0,0 +1,336 @@
...
+ *
+ * Anonymity behaviour is as follows:
+ *
+ * - If Posthog isn't configured in `config.json`, events are not sent.
+ * - If [Do Not Track](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/doNotTrack) is
+ *   enabled, events are not sent (this detection is built into posthog and turned on via the
+ *   `respect_dnt` flag being passed to `posthog.init`).
+ * - If the posthog analytics are explicitly activated by the user in the element call settings,
+ *   a randomised analytics ID is created and stored in account_data for that user (shared between devices)
+ *   so that the user can be identify in posthog.
```
Comment (dbkr) [inline src/PosthogAnalytics.ts:42]:
```suggestion
 *   so that the user can be identified in posthog.
```

**Review by dbkr — APPROVED** [review]
otherwise lgtm - thanks for the detailed comments, they really help!

**Pull request merged**